eth_account/signers/ledger.py

```diff
@@ -0,0 +1,399 @@
+        while True:
+            response = self.unwrapResponseAPDU(0x0101, result, 64)
+            if response is not None:
+                result = response
+                dataStart = 0
+                swOffset = len(response) - 2
+                dataLength = len(response) - 2
+                self.device.set_nonblocking(True)
+                break
+            result.extend(bytearray(self.device.read(65)))
+        sw = (result[swOffset] << 8) + result[swOffset + 1]
+        response = result[dataStart : dataLength + dataStart]
+        if self.debug:
+            print("HID <= %s  %.2x" % (to_hex(response), sw))
+        if sw != 0x9000:
+            possibleCause = "Unknown reason"
+            if sw == 0x6982:
+                possibleCause = "Have you uninstalled the existing CA with resetCustomCA first?"
+            if sw == 0x6985:
+                possibleCause = "Condition of use not satisfied (denied by the user?)"
+            if sw == 0x6a84 or sw == 0x6a85:
+                possibleCause = "Not enough space?"
+            if sw == 0x6484:
+                possibleCause = "Are you using the correct targetId?"
+            raise Exception("Invalid status %04x (%s)" % (sw, possibleCause))
+        return response
+
+    def waitFirstResponse(self, timeout):
+        start = time.time()
+        data = ""
+        while len(data) == 0:
+            data = self.device.read(65)
+            if not len(data):
+                if time.time() - start > timeout:
+                    raise Exception("Timeout")
+                time.sleep(0.0001)
+        return bytearray(data)
+
+    def wrapCommandAPDU(self, channel, command, packetSize):
+        if packetSize < 3:
+            raise Exception("Can't handle Ledger framing with less than 3 bytes for the report")
+        sequenceIdx = 0
+        offset = 0
+        result = struct.pack(">H", channel)
+        extraHeaderSize = 2
+        result += struct.pack(">BHH", 0x05, sequenceIdx, len(command))
+        sequenceIdx = sequenceIdx + 1
+        if len(command) > packetSize - 5 - extraHeaderSize:
+            blockSize = packetSize - 5 - extraHeaderSize
+        else:
+            blockSize = len(command)
+        result += command[offset : offset + blockSize]
+        offset = offset + blockSize
+        while offset != len(command):
+            result += struct.pack(">H", channel)
+            result += struct.pack(">BH", 0x05, sequenceIdx)
+            sequenceIdx = sequenceIdx + 1
+            if (len(command) - offset) > packetSize - 3 - extraHeaderSize:
+                blockSize = packetSize - 3 - extraHeaderSize
+            else:
+                blockSize = len(command) - offset
+            result += command[offset : offset + blockSize]
+            offset = offset + blockSize
+        while (len(result) % packetSize) != 0:
+            result += b"\x00"
+        return bytearray(result)
+
+    def unwrapResponseAPDU(self, channel, data, packetSize):
```

Comment:
Same with this method.  No `self` usage so it should be able to be extracted into a utility method and tested in isolation.